### PR TITLE
Update btcpayserver, nbxplorer

### PR DESCRIPTION
**Pull request by erikarvstedt**
The `btcpayserver` pkg is now built in the same way as the `nbxplorer` pkg:
Use `dotnet publish` instead of `dotnet build` and wrap the executable.

Both packages pass [nix-bitcoin](https://github.com/fort-nix/nix-bitcoin)'s NixOS VM test suite.

To run the test, use the following script.
It leaves no traces (outside of `/nix/store`) on the host system.

```bash
#!/usr/bin/env bash

mkdir -p /tmp/btcpayserver-test/{nixpkgs,nix-bitcoin}
trap "rm -rf /tmp/btcpayserver-test" EXIT
cd /tmp/btcpayserver-test

fetchRepo() { curl -SL https://github.com/$1.tar.gz | tar xz --strip-components=1 -C $2; }
# this PR
fetchRepo erikarvstedt/nixpkgs/archive/ae1716cb7e731f3bf5fb677d1709b9cdbca1d480 nixpkgs
# master as of 2020-11-19
fetchRepo fort-nix/nix-bitcoin/archive/7e81071d0bb2a4cc814660dd4926359c55bdff1a nix-bitcoin

cat > scenarios.nix <<EOF
{ testEnv, ... }:
let
  pkgs = import $(pwd)/nixpkgs {};
in rec {
  btcpayserverUpdate = {
    services.btcpayserver.enable = true;
    services.nbxplorer.package = pkgs.nbxplorer;
    services.btcpayserver.package = pkgs.btcpayserver;
  };
  btcpayserverUpdateRegtest.imports = [
    btcpayserverUpdate
    testEnv.scenarios.regtestBase
  ];
}
EOF
scenarioOverridesFile=$(realpath scenarios.nix) nix-bitcoin/test/run-tests.sh -s btcpayserverUpdate
scenarioOverridesFile=$(realpath scenarios.nix) nix-bitcoin/test/run-tests.sh -s btcpayserverUpdateRegtest
```

### CODE REVIEW

**Comment by kcalvinalvin** [comment]
Tested the following: check hashes, build, run server, check webfront. All looks good

The new btcpayserver script looks good as well :)

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104313` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>btcpayserver</li>
    <li>nbxplorer</li>
  </ul>
</details>

File: pkgs/applications/blockchains/btcpayserver/default.nix [hunk 1]
```diff
@@ -37,21 +35,15 @@ stdenv.mkDerivation rec {
...
 
     dotnet restore --source $TMP/nuget BTCPayServer/BTCPayServer.csproj
-    dotnet build -c Release BTCPayServer/BTCPayServer.csproj
-  '';
-
-  runScript =  ''
-    #!${bash}/bin/bash
-    DOTNET_CLI_TELEMETRY_OPTOUT=1 exec ${dotnetSdk}/bin/dotnet run --no-launch-profile --no-build \
-      -c Release -p @@SHARE@@/BTCPayServer/BTCPayServer.csproj -- "$@"
+    dotnet publish --no-restore --output $out/share/$pname -c Release BTCPayServer/BTCPayServer.csproj
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/btcpayserver/default.nix:38]:
```suggestion
    DOTNET_CLI_TELEMETRY_OPTOUT=1 dotnet publish --no-restore --output $out/share/$pname -c Release BTCPayServer/BTCPayServer.csproj
```

Comment (erikarvstedt) [reply pkgs/applications/blockchains/btcpayserver/default.nix:38]: `DOTNET_CLI_TELEMETRY_OPTOUT` is already set [here](https://github.com/erikarvstedt/nixpkgs/blob/update-btcpayserver/pkgs/applications/blockchains/btcpayserver/default.nix#L32).

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 104313 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>2 packages built:</summary>  <ul>    <li>btcpayserver</li>    <li>nbxplorer</li>  </ul></details>

**Pull request merged**